src/utilities/strategy-helpers/blue-green-helper.ts

```diff
@@ -0,0 +1,328 @@
+    }
+}
+
+
+export function deleteWorkloadsWithLabel(kubectl: Kubectl, deleteLabel: string, deploymentEntityList: any[]) {
+    let delList = []
+    deploymentEntityList.forEach((inputObject) => {
+        const name = inputObject.metadata.name;
+        const kind = inputObject.kind;
+        if (deleteLabel === NONE_LABEL_VALUE) {
+            // if dellabel is none, deletes stable deployments
+            const tempObject = { name : name, kind : kind};
+            delList.push(tempObject);
+        } else {
+            // if dellabel is not none, then deletes new green deployments
+            const tempObject = { name : name+BLUE_GREEN_SUFFIX, kind : kind };
+            delList.push(tempObject);
+        }
+    });
+
+    // deletes the deployments
+    delList.forEach((delObject) => {
+        try {
+            const result = kubectl.delete([delObject.kind, delObject.name]);
+            checkForErrors([result]);
+        } catch (ex) {
+            // Ignore failures of delete if doesn't exist
+        }
+    });
+}
+
+export function cleanUp(kubectl: Kubectl, deploymentEntityList: any[], serviceEntityList: any[]) {
+    // checks if services has some stable deployments to target or deletes them too
+    let delList = []; 
+    serviceEntityList.forEach((inputObject) => {
+        deploymentEntityList.forEach((depObject) => {
+            const kind = depObject.kind;
+            const name = depObject.metadata.name;
+            const serviceSelector: string = getServiceSelector(inputObject);
+            const matchLabels: string = getDeploymentMatchLabels(depObject); 
+            if (!!serviceSelector && !!matchLabels && isServiceSelectorSubsetOfMatchLabel(serviceSelector, matchLabels)) {
+                const existingDeploy = fetchResource(kubectl, kind, name);
+                // checking if it has something to target
+                if (!existingDeploy) {
+                    const tempObject = { name : inputObject.metadata.name, kind : inputObject.kind };
+                    delList.push(tempObject);
+                } 
+            }
+        });
+    });
+
+    delList.forEach((delObject) => {
+        try {
+            const result = kubectl.delete([delObject.kind, delObject.name]);
+            checkForErrors([result]);
+        } catch (ex) {
+            // Ignore failures of delete if doesn't exist
+        }
+    });
+}
+
+export function deleteWorkloadsAndServicesWithLabel(kubectl: Kubectl, deleteLabel: string, deploymentEntityList: any[], serviceEntityList: any[]) {
+    // need to delete services and deployments
+    const deletionEntitiesList = deploymentEntityList.concat(serviceEntityList);
+    let deleteList = []
+    deletionEntitiesList.forEach((inputObject) => {
+        const name = inputObject.metadata.name;
+        const kind = inputObject.kind;
+        if (!deleteLabel) {
```

Comment:
Make use of `deleteLabel` in this method consistent with `deleteWorkloadsWithLabel`